### ISSUE TITLE
Bad character in a comment will break require()

### ISSUE THREAD

**Issue by NathanaelA** [NativeScript/ios-runtime#450]
I have some code that ran on Android fine; pulled it down from the github repo to my Mac; attempted to run it and it kept crashing saying the variable wasn't defined.    Example:

```
var Me = require('./test.js');   
// makes Me === {}  if test.js contains a comment anywhere in it with the following: 
// ... don’t  ...
```
Some how the ' in don't is actually a high ascii character (0x92).  
This character causes the ios runtime to NOT return or execute the required file; and their is NO error!!!
Which of course lets your application to crash elsewhere in the process when it tries to do something with that required file...

Please note the required FILE actually doesn't run, I put a console.log in it at the very top of it, and it never outputted...

**Comment by jasssonpet** [comment]
Hello, @NathanaelA,

I've tried reproducing this odd behavior with the following files:

```test.js
exports.x = 3;
// ... don’t  ...
exports.y = 4;
```

```index.js
var Me = require('./test.js');
console.dir(Me);
```

But it printed `x: 3, y: 4`. Maybe the special character was lost during copying. Can you confirm that we are talking about the `\u2019` character. Also if you could send me this as files, that would really help reproducing the issue.

Thank you!

**Comment by NathanaelA** [comment]
Yeah, attached is a full test case. The only thing you need to do is install the ios runtimes.  It has everything else.  ;-)     I output everything to the console; and you will see the console in the test_bad.js doesn't run, nor does the module.exports get assigned.  
[testChar.zip](https://github.com/NativeScript/ios-runtime/files/70549/testChar.zip)

**Comment by NathanaelA** [comment]
Oh and I realize what the character is; once I found it; it was simple to remove.  But it took a while to try and figure out why it wasn't working as this code had sat a while and I hadn't tested it since way before I did a major upgrade from Maveriks to El Capitan.     So I got to play process of elimination and I never expected the character to cause the iOS runtime to go wonky.

**Comment by jasssonpet** [comment]
Indeed, saving the file with Windows-1252 encoding breaks the runtime.

I'll take mark it as a bug and we'll take a look at it.

**Comment by fealebenpae** [comment]
@jasssonpet: the runtime parses modules as UTF-8 because we use NSString to read the file underneath. Incidentally, node.js [also parses modules as UTF-8](https://github.com/nodejs/node/blob/master/lib/module.js#L414). Most probably `+[NSString stringWithContentsOfFile:encoding:error:]` raises an error when it can't read the file as an UTF-8 string [but we ignore it](https://github.com/NativeScript/ios-runtime/blob/master/src/NativeScript/require.js#L127) and load the module from an empty buffer.

We could try and experiment with `+[NSString stringWithContentsOfFileAtPath:usedEncoding:error:]` in order to support encodings other than UTF-8, but I'd rather we didn't try to do that since node.js doesn't seem to bother, either. We should, however, error out instead of silently loading an empty module. Would that be agreeable, @NathanaelA?

**Comment by jasssonpet** [comment]
@fealebenpae node.js is handling just fine, as @NathanaelA said.

**Comment by fealebenpae** [comment]
@jasssonpet: Apologies, I'm not that proficient in node.js, I assumed from reading the source it supported UTF-8 only. And I'm not familiar enough with Android to be able to say how well it handles charset detection. Like I said, we cause try and use `+[NSString stringWithContentsOfFileAtPath:usedEncoding:error:]` to load the source code of a module, but I haven't used charset detection on iOS and I'm not certain now reliable it is.

**Comment by jasssonpet** [comment]
We'll figure it out :wink:

**Comment by NathanaelA** [comment]
@fealebenpae I just want something to occur.  Either work, since the character is in comments, or fail with an error message.   I really hate silent errors that I have to figure out.  :grin:

**Comment by fealebenpae** [comment]
Resolved in #457
